src/export/ExportUtils.cpp

```diff
@@ -59,3 +54,13 @@ void ExportUtils::InitProgress(std::unique_ptr<BasicUI::ProgressDialog> &pDialog
    return InitProgress(
       pDialog, Verbatim( title.GetName() ), message );
 }
+
+//TODO: used in many places in anticipation that Exporter yields same result, fix that
+TrackIterRange<const WaveTrack> ExportUtils::FindExportWaveTracks(const TrackList& tracks, bool selectedOnly)
+{
+   bool anySolo = !(( tracks.Any<const WaveTrack>() + &WaveTrack::GetSolo ).empty());
+
+   return tracks.Any<const WaveTrack>()
+            + ( selectedOnly ? &Track::IsSelected : &Track::Any  )
+            - ( anySolo ? &WaveTrack::GetNotSolo :  &WaveTrack::GetMute);
```

Comment:
`Track` which is the base of `SampleTrack` does expose selectedness.

But it is `PlayableTrack` that defines mute and solo and it is not a base of `SampleTrack`.

Yet you still might apply `Filter` to the track range to get a range of `SampleTrack`.

Maybe `PlayableTrack` should instead become a mix-on property of tracks that you discover with a `dynamic_cast`?  That's a bigger idea I should think about for future.

Reply:
Feel free to ignore all the suggestions about `SampleTrack` for a draft.  But remember the idea.